src/ops.rs

```diff
@@ -27,6 +27,24 @@ pub fn check_command_exists(command: &str) -> Result<()> {
     }
 }
 
+pub fn check_file_modified(file: &str) -> Result<bool> {
+    trace!("check_file_modified({file})");
+    debug!("Checking if {file} is modified");
+
+    // Git returns 0 if the file is not modified, 1 if it is
+    let is_dirty = Command::new("git")
+        .args(["status", "--porcelain", "|", "grep", file])
+        .output()?;
```

Comment:
Does this actually work? I would suggest using `Stdio::piped()` to send output from one command to another. [Example](https://chat.openai.com/share/99e1658c-8ca4-47eb-a7e2-3a89608c7fca) from ChatGPT (don't use the `expect` plz).